### PR TITLE
Issue 41: Add support for multiple Bookie ledger directories 

### LINKED ISSUE #41: Add support for multiple Bookie ledger directories
### Description
Bookkeeper supports writing to multiple ledger directories for better performance. The ledgers should each be a separate volume and supplied to bookie with ledgerDirs which is a comma separated list of mounted directories.

### Importance
must-have

### Location
https://github.com/pravega/bookkeeper-operator/blob/master/pkg/controller/bookkeepercluster/bookie.go#L133

### PR DESCRIPTION
**Pull request by anishakj**
Signed-off-by: anisha.kj <anisha.kj@dell.com>

### Change log description

Ledger and journal directories were hard coded in bookkeeper pod spec. Made changes to configure multiple journal/ledger directories if it is provided as part of `Options`

### Purpose of the change

Fixes #41 

### What the code does
If journal/ledger directories are specified as part of Options that will be used as Mountpath. Otherwise default will be used

### How to verify it

Tested by giving ledger/journal directories in bookkeeper spec options and with out giving the directory path

### CODE REVIEW

File: pkg/controller/bookkeepercluster/bookie.go [hunk 1]
```diff
@@ -113,6 +113,16 @@ func makeBookiePodSpec(bk *v1alpha1.BookkeeperCluster) *corev1.PodSpec {
 			},
 		})
 	}
+	ledgerDirs := "/bk/ledgers"
+	journalDirs := "/bk/journal"
+	if bk.Spec.Options["ledgerDirectories"] != "" {
```
Comment (pbelgundi) [inline pkg/controller/bookkeepercluster/bookie.go:118]: User may not provide "ledgerDirectories" in Bookkeeper Options.
So need to add a key exists check in the map, using:
```
if ledgerDirs, ok := bk.Spec.Options["ledgerDirectories"]; ok {
    //ledgerDriectories propery is provided by user, ... so do blah blah blah...
}
```

File: pkg/controller/bookkeepercluster/bookie.go [hunk 2]
```diff
@@ -130,11 +140,11 @@ func makeBookiePodSpec(bk *v1alpha1.BookkeeperCluster) *corev1.PodSpec {
...
-						MountPath: "/bk/ledgers",
+						MountPath: ledgerDirs,
 					},
 					{
 						Name:      JournalDiskName,
-						MountPath: "/bk/journal",
+						MountPath: journalDirs,
 					},
 					{
 						Name:      IndexDiskName,
```
Comment (pbelgundi) [inline pkg/controller/bookkeepercluster/bookie.go:150]: User may want to have a different mount path for IndexDirectories as wel.
Lets have the same check for IndexDirectories too...

File: pkg/controller/bookkeepercluster/bookie.go [hunk 3]
```diff
@@ -113,6 +113,16 @@ func makeBookiePodSpec(bk *v1alpha1.BookkeeperCluster) *corev1.PodSpec {
...
 		})
 	}
+	ledgerDirs := "/bk/ledgers"
+	journalDirs := "/bk/journal"
+	if bk.Spec.Options["ledgerDirectories"] != "" {
+
+		ledgerDirs = bk.Spec.Options["ledgerDirectories"]
+
+	}
+	if bk.Spec.Options["journalDirectories"] != "" {
```
Comment (pbelgundi) [inline pkg/controller/bookkeepercluster/bookie.go:123]: Would be good to add UTs to cover this code.

Comment (anishakj) [reply pkg/controller/bookkeepercluster/bookie.go]: Done

File: pkg/controller/bookkeepercluster/bookie.go (hunk 2 above)
Comment (anishakj) [reply pkg/controller/bookkeepercluster/bookie.go:163]: Done

File: pkg/controller/bookkeepercluster/bookie.go [hunk 4]
```diff
@@ -114,6 +114,27 @@ func makeBookiePodSpec(bk *v1alpha1.BookkeeperCluster) *corev1.PodSpec {
 		})
 	}
 
+	var ledgerDirs, journalDirs, indexDirs string
+	var ok bool
+
+	if ledgerDirs, ok = bk.Spec.Options["ledgerDirectories"]; ok {
```
Comment (pbelgundi) [inline pkg/controller/bookkeepercluster/bookie.go:120]: Can be written more concisely as 
```
if ledgerDirs, ok = bk.Spec.Options["ledgerDirectories"]; !ok {
// default value if user did not set ledgerDirectories in options 
ledgerDirs = "/bk/ledgers"
}
```

Comment (RaulGracia) [reply pkg/controller/bookkeepercluster/bookie.go:120]: Agree with @pbelgundi.

Comment (anishakj) [reply pkg/controller/bookkeepercluster/bookie.go]: Changed

**Comment by anishakj** [comment]
> LGTM. Lets make sure I/O works fine with ledgerDirectories/journalDirectories provided in bookkeeper Options and with default value

Verified that I/O is working fine in both cases.

**Pull request merged**